core/src/main/scala/kafka/log/Log.scala

```diff
@@ -565,9 +555,19 @@ class Log(@volatile var dir: File,
     recoveryPoint
   }
 
-  private def loadProducerState(lastOffset: Long, reloadFromCleanShutdown: Boolean): Unit = lock synchronized {
+  private def loadProducerState(lastOffset: Long,
+                                reloadFromCleanShutdown: Boolean,
+                                producerStateManager: ProducerStateManager): Unit = lock synchronized {
     checkIfMemoryMappedBufferClosed()
     val messageFormatVersion = config.messageFormatVersion.recordVersion.value
+    val segments = logSegments
+    val offsetsToSnapshot =
+      if (segments.size > 0) {
+        val nextLatestSegmentBaseOffset = lowerSegment(segments.last.baseOffset).map(_.baseOffset)
+        Seq(nextLatestSegmentBaseOffset, Some(segments.last.baseOffset), Some(lastOffset))
+      } else {
+        Seq(Some(lastOffset))
+      }
     info(s"Loading producer state from offset $lastOffset with message format version $messageFormatVersion")
```

Comment:
Can we fix this message? It seems a little misleading since we are really loading the producer state up to the offset.